File: C#/Projects/SnomedQuery.Parser/RF2RelationshipGroup.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SnomedQuery.Parser
{
  /// <summary>
  /// Group of relationships, all with same SNOMED id.
  /// </summary>
  public class RF2RelationshipGroup : TRF2ItemGroup<RF2Relationship>
  {
    /// <summary>
    /// Constructor
    /// </summary>
    public RF2RelationshipGroup()
    {
    }

    /// <summary>
    /// Add relationsip to relationship group.
    /// </summary>
    /// <param name="relationship"></param>
    public void AddRelationship(RF2Relationship relationship)
    {
      this.items.Add(relationship);
    }

    /// <summary>
    /// Override ToString() method.
    /// </summary>
    /// <returns></returns>
    public override string ToString()
    {
      StringBuilder sb = new StringBuilder();
      sb.Append($"RF2RelationshipGroup [{Id}] ");
      RF2Relationship active = this.items[0];
      sb.Append($"Type [{active.TypeId}] '{active.TypeConcept.GetFullySpecifiedName()}'");
      sb.Append($"Source [{active.SourceId}] '{active.Source.GetFullySpecifiedName()}'");
      sb.Append($"Dest [{active.DestinationId}] '{active.Destination.GetFullySpecifiedName()}'");

      return sb.ToString();
    }
  }
}

File: Java/SnomedQuery/Parser/RF2RelationshipGroup.java
package Parser;

/**
 * <p>
 * Group of relationships, all with same SNOMED id.
 * </p>
 * 
 * @author Travis Lukach
 */
public class RF2RelationshipGroup extends TRF2ItemGroup<RF2Relationship> {
	/**
	 * <p>
	 * Constructor
	 * </p>
	 */
	public RF2RelationshipGroup() {
	}

	/**
	 * <p>
	 * Add relationsip to relationship group.
	 * </p>
	 * 
	 * @param relationship
	 */
	public final void addRelationship(RF2Relationship relationship) {
		this.items.add(relationship);
	}

	/**
	 * <p>
	 * Override ToString() method.
	 * </p>
	 * 
	 * @return
	 */
	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append(String.format("RF2RelationshipGroup [%1$s] ", getId()));
		RF2Relationship active = this.items.get(0);
		sb.append(String.format("Type [%1$s] '%2$s'", active.getTypeId(),
				active.getTypeConcept().getFullySpecifiedName()));
		sb.append(String.format("Source [%1$s] '%2$s'", active.getSourceId(),
				active.getSource().getFullySpecifiedName()));
		sb.append(String.format("Dest [%1$s] '%2$s'", active.getDestinationId(),
				active.getDestination().getFullySpecifiedName()));

		return sb.toString();
	}
}